torch/_higher_order_ops/wrap.py

```diff
@@ -25,45 +29,34 @@ def wrapper():
 
 wrap = Wrap()
 
-class WrapActivationCheckpoint(HigherOrderOperator):
+class HandleActivationCheckpoint(HigherOrderOperator):
     """
-    This operator is used to wrap torch.utils.checkpoint. This avoids
+    This operator is supposed to be used only with torch.compile stack.
+    There are two modes in this operator:
+
+    Mode 1: WrapActivationCheckpoint
+    This mode is used for selective checkpointing + torch.compile.
+    Under this mode, we wrap torch.utils.checkpoint. This avoids
     TorchDynamo to look into saved tensor hooks and directly passes the control
     to AOT Autograd, which is ok with tracing saved tensor hooks. As a result of
-    AOT tracing torch.utils.checkpoint code, we have a backward graph with
-    recomputed forward nodes.
+    AOT tracing torch.utils.checkpoint code, particularly _CachingTorchDispatchMode
+    and _CachedTorchDispatchMode, we obtain FX graph with added "recomputable" tag
+    to the nodes that should be recomputed. Then, similar to TagActivationCheckpoint
+    case, we rely on the partitioners to actually duplicate the nodes.
```

Comment:
This is confusing. Why do we even need `recomputable` tags if we are relying on AOT Autograd to do the tracing.

Imo, we should keep WrapActivationCheckpoint simple - let AOT Autograd take care of everything. In fact, this TorchDispatch facility seems to sit well with TagActivationCheckpoint to me.